### PR TITLE
dev-db/mariadb-connector-odbc:  Initial submission.

### PR DESCRIPTION
**Pull request by jkroonza**
Signed-off-by: Jaco Kroon <jaco@uls.co.za>

### CODE REVIEW

**Comment by gyakovlev** [comment]
@gentoo/mysql  wanna chime in?

File: dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Copyright 2018 Jaco Kroon
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit cmake-utils multilib-minimal
```
Comment (mgorny) [inline dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild]: = cmake-multilib

File: dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+DESCRIPTION="MariaDB Connector/ODBC"
+HOMEPAGE="https://downloads.mariadb.org/connector-odbc/"
+SRC_URI="https://downloads.mariadb.org/interstitial/connector-odbc-${PV}/${P}-ga-src.tar.gz"
+
+LICENSE="LGPL-2.1"
```
Comment (mgorny) [inline dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild]: You sure it's not 2.1+? (`or newer`)

File: dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+SLOT="0/3"
+KEYWORDS="~amd64 ~x86"
+IUSE="ssl gnutls test"
```
Comment (mgorny) [inline dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild]: Sort flags lexically

File: dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+S="${S}-ga-src"
+
+DEPEND="dev-db/mariadb-connector-c"
```
Comment (mgorny) [inline dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild]: `:=` surely?

File: dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+RDEPEND="${DEPEND}"
+BDEPEND=""
+
+multilib_src_configure() {
+	append-cppflags $(mariadb_config --cflags)
+	local mycmakeargs=(
+	-DWITH_SSL:STRING=$(usex ssl $(usex gnutls GNUTLS OPENSSL) OFF)
```
Comment (mgorny) [inline dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild]: Indent.

File: dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild (hunk 3 above)
Comment (mgorny) [inline dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild]: Also, you need `RESTRICT="!test? ( test )"` to disable tests with USE=-test.

File: dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+	-DWITH_UNIT_TESTS=$(usex test ON OFF)
+	-DMARIADB_LINK_DYNAMIC=YES
+	-DUSE_SYSTEM_INSTALLED_LIB=YES
+	#-DCMAKE_C_FLAGS="$(mariadb_config --cflags)"
```
Comment (mgorny) [inline dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild]: What's this?

File: dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+	cmake-utils_src_compile
+}
+
+multilib_src_install() {
+	cmake-utils_src_install
+}
+
+multilib_src_install_all() {
+	dodir /usr/share/${PN}
+	sed -e "s,lib/lib,$(get_libdir)/lib,g" < "${FILESDIR}/odbcinst.ini" > "${D}/usr/share/${PN}/odbcinst.ini"
```
Comment (mgorny) [inline dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild]: Given how tiny the file is, maybe inline it here using `newins - ...` and a heredoc.

File: dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+pkg_config() {
+	[ "${ROOT}" != "/" ] && die "Sorry, non-standard ROOT setting is not supported."
```
Comment (mgorny) [inline dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild]: Use `[[ ... ]]` in ebuilds, and then quoting becomes unnecessary.

File: dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+	if /usr/bin/odbcinst -q -d -n maodbc &>/dev/null; then
```
Comment (mgorny) [inline dev-db/mariadb-connector-odbc/mariadb-connector-odbc-3.0.6.ebuild]: Why the full path?

**Comment by mgorny** [comment]
Ping.

**Pull request closed without merging**